### ISSUE TITLE
operator remove: with multiple revision fails with Operator controller is not installed

### ISSUE THREAD

**Issue by shamsher31** [istio/istio.io#9171]
Ref: https://istio.io/latest/docs/setup/install/operator/#uninstall

If multiple revision of Istio is installed with the operator. During the previous revision cleanup using the following command.

```
kubectl delete istiooperators.install.istio.io -n istio-system example-istiocontrolplane
```

The `operator remove` command with revision is not able to find a controller for the operator and throws the following error.
```
istioctl operator remove --revision 1-8-1
Operator controller is not installed in istio-operator namespace (no Deployment detected).
Aborting, use --force to override.
```

Found this issue while debugging https://github.com/istio/istio/issues/31304

**Comment by shamsher31** [comment]
cc: @morvencao

**Comment by tanjunchen** [comment]
@morvencao 
In this case, you can try to delete the new iop, and you find that the iop is deleted, but the istio cluster is not deleted. An error occurred during the synchronization phase of the istio operator. see *API server conflict during finalizer removal, retrying.* 
 https://github.com/istio/istio/pull/30949  will  fix it?

**Comment by morvencao** [comment]
> In this case, you can try to delete the new iop, and you find that the iop is deleted, but the istio cluster is not deleted.

Haven't try this case, you delete new IOP to rollback?

**Comment by tanjunchen** [comment]
no, I just want to uninstall the  istio cluster. you can try to do this.
